Client/src/Components/EventCreation/EventForm.jsx

```diff
@@ -0,0 +1,173 @@
+import { useState, useContext} from 'react';
+import { UserIdContext } from '../UserEvents/UserIdContext';
+import FetchAltText from './FetchAltText';
+import { AltTextContext } from './AltTextContext';
+import './CreateEvent.css';
+
+function EventForm() {
+    const { userId } = useContext(UserIdContext);
+    const { eventAltText} = useContext(AltTextContext);
+    const [eventId, setEventId] = useState('');
+    const [eventTitle, setEventTitle] = useState(''); 
+    const [date, setDate] = useState(''); 
+    const [location, setLocation] = useState(''); 
+    const [eventType, setEventType] = useState(''); 
+    const [eventGroup, setEventGroup] = useState('');
+    const [eventDescription, setEventDescription] = useState(''); 
+    const [error, setError] = useState('');
+    const [isFormVisible, setIsFormVisible] = useState(true);
+
+    // Handle photo upload
+    const handlePhotoUpload = async (event) => {
+        const file = event.target.files[0];
+        if (!file) return;
+
+        const uploadData = new FormData();
+        uploadData.append('file', file);
+
+        try {
+            const response = await fetch('/uploadPhoto', {
+                method: 'POST',
+                body: uploadData,
+            });
+
+            if (!response.ok) {
+                throw new Error(`HTTP error! Status: ${response.status}`);
+            }
+
+            const responseData = await response.json();
+            setEventId(responseData.eventid);
+        } catch (error) {
+            console.error('Error uploading photo:', error);
+            setError(error.message);
+        }
+    };
+
+    // Handle form submission (PUT request to update the event)
+    const handleSubmitPut  = async (event) => {
+        event.preventDefault();
+
+        const putRequestBody = {
+            userId,
+            date,
+            location,
+            eventType,
+            eventDescription,
+            eventTitle,
+            eventAltText,
+            eventGroup,
+        };
+
+        try {
+            const response = await fetch(`http://localhost:1113/editEvents/${eventId}`, {
```

Comment:
You could create an environment variable to handle your links, so you can just have /editEvents/${eventId}. This would help in preparing your app for production.

Reply:
Fixed and made updated changes!

Reply:
https://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/